sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySchemaTransformReadProvider.java

```diff
@@ -0,0 +1,176 @@
+  }
+
+  /**
+   * Implementation of the {@link TypedSchemaTransformProvider} outputCollectionNames method. Since
+   * a single output is expected, this returns a list with a single name.
+   */
+  @Override
+  public List<String> outputCollectionNames() {
+    return Collections.singletonList(OUTPUT_TAG);
+  }
+
+  /**
+   * An implementation of {@link SchemaTransform} for BigQuery read jobs configured using {@link
+   * BigQuerySchemaTransformReadConfiguration}.
+   */
+  static class BigQuerySchemaTransformRead implements SchemaTransform {
+    private final BigQuerySchemaTransformReadConfiguration configuration;
+
+    BigQuerySchemaTransformRead(BigQuerySchemaTransformReadConfiguration configuration) {
+      this.configuration = configuration;
+    }
+
+    /** Implements {@link SchemaTransform} buildTransform method. */
+    @Override
+    public PTransform<PCollectionRowTuple, PCollectionRowTuple> buildTransform() {
+      return new BigQuerySchemaTransformReadTransform(configuration);
+    }
+  }
+
+  /**
+   * An implementation of {@link PTransform} for BigQuery read jobs configured using {@link
+   * BigQuerySchemaTransformReadConfiguration}.
+   */
+  static class BigQuerySchemaTransformReadTransform
+      extends PTransform<PCollectionRowTuple, PCollectionRowTuple> {
+    private final BigQuerySchemaTransformReadConfiguration configuration;
+
+    BigQuerySchemaTransformReadTransform(BigQuerySchemaTransformReadConfiguration configuration) {
+      this.configuration = configuration;
+    }
+
+    @Override
+    public PCollectionRowTuple expand(PCollectionRowTuple input) {
+      if (!input.getAll().isEmpty()) {
+        throw new IllegalArgumentException(
+            String.format(
+                "%s %s input is expected to be empty",
+                input.getClass().getSimpleName(), getClass().getSimpleName()));
+      }
+      PCollection<TableRow> tableRowPCollection = input.getPipeline().apply(toTypedRead());
+      Schema schema = tableRowPCollection.getSchema();
+      PCollection<Row> rowPCollection =
+          tableRowPCollection.apply(
+              MapElements.into(TypeDescriptor.of(Row.class))
+                  .via((tableRow) -> BigQueryUtils.toBeamRow(schema, tableRow)));
+      return PCollectionRowTuple.of(OUTPUT_TAG, rowPCollection);
+    }
+
+    BigQueryIO.TypedRead<TableRow> toTypedRead() {
+      JobType jobType = configuration.getJobType();
+      switch (jobType) {
+        case QUERY:
+          return toQueryTypedRead();
+
+        case EXTRACT:
+          return toExtractTypedRead();
+
+        default:
+          throw new InvalidConfigurationException(
+              String.format("invalid job type for BigQueryIO read, got: %s", jobType));
+      }
+    }
+
+    private BigQueryIO.TypedRead<TableRow> toExtractTypedRead() {
+      return BigQueryIO.readTableRowsWithSchema().from(configuration.getTableSpec());
+    }
+
+    private BigQueryIO.TypedRead<TableRow> toQueryTypedRead() {
```

Comment:
This code throws the following error when implementing based on [this comment](https://github.com/apache/beam/pull/17181#discussion_r835454867)

```
/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_PVR_Flink_Docker_Commit/src/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySchemaTransformReadProvider.java:155: error: [argument.type.incompatible] incompatible argument for parameter tableSpec of from.
13:26:39       return BigQueryIO.readTableRowsWithSchema().from(configuration.getTableSpec());
13:26:39                                                                                  ^
13:26:39   found   : @Initialized @Nullable String
13:26:39   required: @Initialized @NonNull String
13:26:39 /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_PVR_Flink_Docker_Commit/src/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySchemaTransformReadProvider.java:159: error: [argument.type.incompatible] incompatible argument for parameter arg0 of requireNonNull.
13:26:39       String query = Objects.requireNonNull(configuration.getQuery());
13:26:39                                                                   ^
13:26:39   found   : @Initialized @Nullable String
13:26:39   required: @Initialized @NonNull String
13:26:39 /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_PVR_Flink_Docker_Commit/src/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySchemaTransformReadProvider.java:164: error: [argument.type.incompatible] incompatible argument for parameter location of withQueryLocation.
13:26:39         read = read.withQueryLocation(configuration.getQueryLocation());
13:26:39                                                                     ^
13:26:39   found   : @Initialized @Nullable String
13:26:39   required: @Initialized @NonNull String
13:26:39 /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_PVR_Flink_Docker_Commit/src/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySchemaTransformReadProvider.java:168: error: [condition.nullable] condition on a possibly-null value (configuration.getUseStandardSql())
13:26:39         if (configuration.getUseStandardSql()) {
```

Reply:
Would this error be resolved by adding a SuppressWarnings annotation related to https://issues.apache.org/jira/browse/BEAM-10402

Reply:
> Should I add the following annotation?
> 
> ```
> @SuppressWarnings({
>   "nullness" // TODO(https://issues.apache.org/jira/browse/BEAM-10402)
> })
> ```

Well it's calling out a real problem right? The function it's calling doesn't handle null but you might have a null. You should probably use Preconditions.check and fail if it's null and shouldn't be. I would think that might help the check but if not then we can add that suppress too. But I would expect there's a way to indicate to the checker that we have checked for null?